packages/@aws-cdk/aws-apprunner/lib/service.ts

```diff
@@ -0,0 +1,603 @@
+   */
+  readonly serviceStatus: string;
+}
+
+/**
+ * Represents the App Runner Service.
+ */
+export interface IService extends cdk.IResource {
+  /**
+   * The Name of the service.
+   */
+  readonly serviceName: string;
+  /**
+   * The ARN of the service.
+   */
+  readonly serviceArn: string;
+}
+
+/**
+ * The App Runner Service.
+ */
+export class Service extends cdk.Resource {
+  /**
+   * Import from service name.
+   */
+  public static fromServiceName(scope: Construct, id: string, serviceName: string): IService {
+    class Import extends cdk.Resource {
+      public serviceName = serviceName;
+      public serviceArn = cdk.Stack.of(this).formatArn({
+        resource: 'service',
+        service: 'apprunner',
+        resourceName: serviceName,
+      })
+    }
+    return new Import(scope, id);
+  }
+  /**
+   * Import from service attributes.
+   */
+  public static fromServiceAttributes(scope: Construct, id: string, attrs: ServiceAttributes): IService {
+    const serviceArn = attrs.serviceArn;
+    const serviceName = attrs.serviceName;
+    const serviceUrl = attrs.serviceUrl;
+    const serviceStatus = attrs.serviceStatus;
+
+    class Import extends cdk.Resource {
+      public readonly serviceArn = serviceArn
+      public readonly serviceName = serviceName
+      public readonly serviceUrl = serviceUrl
+      public readonly serviceStatus = serviceStatus
+    }
+
+    return new Import(scope, id);
+  }
+  private readonly props: ServiceProps;
+  private accessRole?: iam.IRole;
+  private ecrRepository?: ecr.IRepository;
+  /**
+   * The ARN of the Service.
+   * @attribute
+   */
+  readonly serviceArn: string;
+  /**
+   * The ID of the Service.
+   * @attribute
+   */
+  readonly serviceId: string;
+  /**
+   * The URL of the Service.
+   * @attribute
+   */
+  readonly serviceUrl: string;
+  /**
+   * The status of the Service.
+   * @attribute
+   */
+  readonly serviceStatus: string;
+  /**
+   * The name of the service.
+   * @attribute
+   */
+  readonly serviceName: string;
+  constructor(scope: Construct, id: string, props: ServiceProps = {}) {
+    super(scope, id);
+    // either image or code is allowed. Not both.
+    if ((!props.code && !props.image) || (props.code && props.image)) {
+      throw new Error('Either image or code is required, not both.');
+    }
+    // connection is required if code is defined
+    if (props.code && !props.connection) {
+      throw new Error('connection is required for github repository source.');
+    }
+    this.props = props;
+
+    // generate an IAM role only when ImageRepositoryType is ECR and props.role is undefined
+    this.accessRole = (this.props.image && this.props.image.type == ImageRepositoryType.ECR) ?
+      this.props.accessRole ? this.props.accessRole : this.generateDefaultRole() : undefined;
+
+    const resource = new CfnService(this, 'Resource', {
+      instanceConfiguration: {
+        cpu: props.cpu?.valueOf(),
+        memory: props.memory?.valueOf(),
```

Comment:
I know that for ECS, we did not do this, and there were discussions about it... Not sure if this is a similar enough case that we also want to not do this... It seems to me that you might instead want to expose constants, but let people pass their own values, too, in case new options get added in the future.

Reply:
OK maybe we can provide enum-like class like [KubernetesVersion](https://github.com/aws/aws-cdk/blob/1f23313e889a00d85e7cbc56a64242d6e7762835/packages/%40aws-cdk/aws-eks/lib/cluster.ts#L611)?